### PR TITLE
MLIBZ-2633: .NET Standard 2.0

### PR DESCRIPTION
**Pull request by heyzooi**
#### Description

.NET Standard 2.0

#### Changes

- New projects with new profile

#### Tests

- Same Unit Tests

### CODE REVIEW

**Comment by heyzooi** [comment]
@vinaygahlawat I think now we are ready to merge, please feel free to review the PR

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,40 @@
...
+        - export PATH="$PATH:/usr/local/share/dotnet"
+        - *install_coverlet
+    - name: "Mono"
+      language: csharp
+      solution: Kinvey.sln
+      install:
+        - wget -O pubnub.zip https://www.nuget.org/api/v2/package/PubnubNetPlatform/3.8.7
+        - mkdir packages/pubnubnetplatform.3.8.7
+        - mv pubnub.zip packages/pubnubnetplatform.3.8.7/.
+        - unzip packages/pubnubnetplatform.3.8.7/pubnub.zip -d packages/pubnubnetplatform.3.8.7/
```
[the remark this answers is not in the document]
Comment (heyzooi) [reply .travis.yml]: sure, let me try do that

File: Kinvey.Core/Client/Client.cs [hunk 2]
```diff
@@ -190,10 +190,11 @@ public virtual Client Build()
 				Client c =  new Client(this.HttpRestClient, this.BaseUrl, this.ServicePath, this.RequestInitializer, this.Store);
 //				c.offline_platform = this.offlinePlatform;
 //				c.filePath = this.filePath;
-				c.CacheManager = this.CacheManager;
+                c.CacheManager = this.CacheManager;
 				c.logger = this.log;
 				c.senderID = this.senderID;
 				c.SSOGroupKey = this.ssoGroupKey;
+                if (!string.IsNullOrEmpty(this.MICHostName)) c.MICHostName = this.MICHostName;
```
[the remark this answers is not in the document]
Comment (heyzooi) [reply Kinvey.Core/Client/Client.cs]: this was needed to be possible to mock responses for the unit tests

File: Kinvey.Core/Utils/HelperMethods.cs [hunk 3]
```diff
@@ -83,27 +85,16 @@ public static string GetRequestStartTime(IRestResponse response)
...
-                        {
-                            XKinveyRequestStart = item.ToString();
-                        }
-                        else if (valueType.IsConstructedGenericType)
+                        var values = item as IEnumerable<string>;
+                        var requestStart = values.FirstOrDefault();
+                        if (requestStart != null)
                         {
-                            if (valueType.Name.Contains("List"))
-                            {
```
[the remark this answers is not in the document]
Comment (heyzooi) [reply Kinvey.Core/Utils/HelperMethods.cs]: `IEnumerable<string>` is a more generic solution. When I was running the unit tests I got some times `string[]` and sometimes `List<string>`. `IEnumerable<string>` works in both situations.

File: Kinvey.Net/Kinvey.Net.csproj [hunk 4]
```diff
@@ -90,5 +90,5 @@
     <None Include="packages.config" />
   </ItemGroup>
   <Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />
-  <Import Project="..\packages\System.Data.SQLite.Core.1.0.105.0\build\net46\System.Data.SQLite.Core.targets" Condition="Exists('..\packages\System.Data.SQLite.Core.1.0.105.0\build\net46\System.Data.SQLite.Core.targets')" />
+  <Import Project="..\packages\System.Data.SQLite.Core.1.0.109.1\build\net46\System.Data.SQLite.Core.targets" Condition="Exists('..\packages\System.Data.SQLite.Core.1.0.109.1\build\net46\System.Data.SQLite.Core.targets')" />
 </Project>
```
[the remark this answers is not in the document]
Comment (heyzooi) [reply Kinvey.Net/Kinvey.Net.csproj]: not really, let me revert this change

File: .travis.yml [hunk 5]
```diff
@@ -28,14 +28,10 @@ matrix:
...
-        - wget -O pubnub.zip https://www.nuget.org/api/v2/package/PubnubNetPlatform/3.8.7
-        - mkdir packages/pubnubnetplatform.3.8.7
-        - mv pubnub.zip packages/pubnubnetplatform.3.8.7/.
-        - unzip packages/pubnubnetplatform.3.8.7/pubnub.zip -d packages/pubnubnetplatform.3.8.7/
         - nuget restore Kinvey.sln
       script:
-        - xbuild Kinvey.Core/Kinvey.Core.csproj
-        - xbuild TestFramework/Tests.Integration/Tests.Integration.csproj
+        - msbuild Kinvey.Core/Kinvey.Core.csproj
+        - msbuild TestFramework/Tests.Integration/Tests.Integration.csproj
```
Comment (vinaygahlawat) [inline .travis.yml]: Is the preferred method here to use `msbuild` instead of `dotnet`?

**Comment by vinaygahlawat** [comment]
With all the removed packages (presumably because they are no longer needed for the .NET Standard library), will this affect how the old projects work?

File: .travis.yml (hunk 5 above)
Comment (heyzooi) [reply .travis.yml]: we are testing `dotnet` as well, but this one here is to test against mono, which apparently the `msbuild` is the preferred way

Comment (heyzooi) [reply .travis.yml]: so basically we are testing in 3 environments for now:
- Linux dotnet
- macOS dotnet
- macOS mono

Ideally we should add more in the future like:
- Windows dotnet
- Xamarin iOS
- Xamarin Android

Comment (heyzooi) [reply .travis.yml]: also `xbuild` is now deprecated... i noticed by looking one of the build logs

Comment (heyzooi) [reply .travis.yml]:
```
>>>> xbuild tool is deprecated and will be removed in future updates, use msbuild instead <<<<
```

Comment (vinaygahlawat) [reply .travis.yml]: Ok good. I knew about `xbuild` but didn't know about `mono` preferring `msbuild`.

**Pull request merged**